### PR TITLE
Retire trimet-mod-otp + Migrate from Mastarm to Craco

### LINKED ISSUE #468: Pattern Viewer station blob connectors not aligned properly
Screen Shot 2021-10-06 at 9 57 17 AM

Seeing this issue only on trimet-mod-otp instances. Has to do with line-height changes. 

Confirmed fixed by https://github.com/opentripplanner/otp-react-redux/pull/447, but the fix is small and can probably be extracted.

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
This PR retires trimet-mod-otp, integrating its functionality into otp-rr.

This PR also adds typescript to files which the new linting rules request it to have.

Blocked waiting for flex PR #470 being merged into dev.

Closes #468

Issues to resolve:

- [x] initial map render doesn't work on desktop
- [x] geocoder broken (related to map render broken?)
- [x] build is broken again due to CSS import (mini-css-extract-plugin ?)
- [x] typing is missing on a lot of files
- [x] update example config should work out of the box -> will fix a11y tests
- [x] update README
- [x] React fast refresh doesn't work (weird version issues)
- [x] source maps not working
- [x] OTP UI deps have to be updated to fix tests
- [ ] test call taker and field trip (need external verification)
- [x] remove superfluous files (general cleanup)
- [x] cleanup web pack config
- [x] deployment

### Notes for reviewer
Until https://github.com/opentripplanner/otp-ui/pull/295 is merged, the location field will continue to not work. The main aspects to test are if everything works as expected using all possible configs, and most crucially that the call taker and field trip still work as expected.

I would also like some feedback as to whether to squash some of these commits, as the PR brings quite a few commits.

Finally, `trimet-mod-otp` ran the trimet config out of the box. Should we ship this config with otp-rr now? At the moment, otp-rr does NOT work "out of the box", whereas `trimet-mod-otp` did.

### CODE REVIEW

**Comment by miles-grant-ibigroup** [comment]
Looks like everything is working. CI will lint any file that was changed in the PR, so need to go through all files touched by the PR and migrate them to typescript.

File: lib/components/app/app-menu.tsx [hunk 1]
```diff
@@ -153,59 +179,62 @@ class AppMenu extends Component {
...
         <SlidingPane
-          from='left'
+          from="left"
           isOpen={isPaneOpen}
           onRequestClose={this._togglePane}
-          width='320px'
+          width="320px"
         >
-          <ul className='app-menu'>
+          <div className="app-menu">
```
Comment (binh-dam-ibigroup) [inline lib/components/app/app-menu.tsx:198]: Why the change to `<div>`?

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
+module.exports = {
+  extends: [
+    'standard',
```
Comment (binh-dam-ibigroup) [inline .eslintrc.js:3]: I thought one of these rules dealt with quote style, but apparently some files got through the net, so can you add a rule to enforce strings in single-quotes, please?

File: lib/components/map/bounds-updating-overlay.js [hunk 3]
```diff
@@ -83,39 +88,50 @@ class BoundsUpdatingOverlay extends MapLayer {
...
+    }
     if (itineraryViewChanged) {
       this._fitItineraryViewToMap(newProps, newItinBounds, map)
     } else if (
       (!oldItinBounds && newItinBounds) ||
       (oldItinBounds && newItinBounds && !oldItinBounds.equals(newItinBounds))
     ) {
       map.fitBounds(newItinBounds, { padding: BOUNDS_PADDING })
-    // Pan to to itinerary leg if made active (clicked); newly active leg must be non-null
+      // Pan to to itinerary leg if made active (clicked); newly active leg must be non-null
```
Comment (binh-dam-ibigroup) [inline lib/components/map/bounds-updating-overlay.js:123]: Move this comment inside `else if` block.

File: lib/components/map/bounds-updating-overlay.js [hunk 4]
```diff
@@ -83,39 +88,50 @@ class BoundsUpdatingOverlay extends MapLayer {
...
+      newProps.activeLeg !== oldProps.activeLeg &&
+      newProps.activeLeg !== null
     ) {
       map.fitBounds(
         getLeafletLegBounds(newProps.itinerary.legs[newProps.activeLeg]),
         { padding: BOUNDS_PADDING }
       )
 
-    // If no itinerary update but from/to locations are present, fit to those
+      // If no itinerary update but from/to locations are present, fit to those
```
Comment (binh-dam-ibigroup) [inline lib/components/map/bounds-updating-overlay.js:134]: Move comment to inside `else if` block below.

File: lib/components/map/bounds-updating-overlay.js [hunk 5]
```diff
@@ -135,24 +151,27 @@ class BoundsUpdatingOverlay extends MapLayer {
...
+        map.fitBounds(
+          [
+            [left, bottom],
+            [right, top]
+          ],
+          { padding: BOUNDS_PADDING }
+        )
       }
-    // If only from or to is set, pan to that
+      // If only from or to is set, pan to that
```
Comment (binh-dam-ibigroup) [inline lib/components/map/bounds-updating-overlay.js:162]: Move comment to inside `else if` block below.

File: lib/components/map/bounds-updating-overlay.js [hunk 6]
```diff
@@ -135,24 +151,27 @@ class BoundsUpdatingOverlay extends MapLayer {
...
     } else if (newFrom && fromChanged) {
       map.panTo([newFrom.lat, newFrom.lon])
     } else if (newTo && toChanged) {
       map.panTo([newTo.lat, newTo.lon])
 
-    // If intermediate place is added, extend bounds.
+      // If intermediate place is added, extend bounds.
```
Comment (binh-dam-ibigroup) [inline lib/components/map/bounds-updating-overlay.js:168]: Move comment to inside `else if` block below.

File: lib/components/map/bounds-updating-overlay.js [hunk 7]
```diff
@@ -135,24 +151,27 @@ class BoundsUpdatingOverlay extends MapLayer {
...
     } else if (newIntermediate && intermediateChanged) {
       const bounds = map.getBounds()
       extendBoundsByPlaces(bounds, newIntermediate)
       map.fitBounds(bounds)
 
-    // Pan to to itinerary step if made active (clicked)
+      // Pan to to itinerary step if made active (clicked)
```
Comment (binh-dam-ibigroup) [inline lib/components/map/bounds-updating-overlay.js:174]: Move comment to inside `else if` block below.

File: lib/components/map/point-popup.tsx [hunk 8]
```diff
@@ -10,17 +14,20 @@ const PopupTitle = styled.div`
...
 `
 
-export default function MapPopup ({
+export default function MapPopup({
   mapPopupLocation,
   onSetLocationFromPopup
-}) {
+}: {
+  mapPopupLocation: { name: string }
+  // TODO: what is this function actually
```
Comment (binh-dam-ibigroup) [inline lib/components/map/point-popup.tsx:22]: This sets the "from" or "to" location when you click anywhere on the map. Comment can be changed to adding types for this method signature.

File: lib/components/narrative/default/tnc-leg.js [hunk 9]
```diff
@@ -13,60 +13,85 @@ const { toSentenceCase } = coreUtils.itinerary
...
+        leg.to.lon
+      }&dropoff[nickname]=${encodeURI(leg.to.name)}`
     }
     const { tncData } = leg
     return (
       <div>
         <p>* estimated travel time does not account for traffic.</p>
         <a
-          className='btn btn-default'
+          className="btn btn-default"
```
Comment (binh-dam-ibigroup) [inline lib/components/narrative/default/tnc-leg.js:51]: Revert conversion to double quotes in this file.

File: lib/components/narrative/narrative-itineraries-header.tsx [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+        flexWrap: 'wrap'
+      }}
+    >
+      {itineraryIsExpanded || showingErrors ? (
+        <>
+          <button
+            className="clear-button-formatting"
+            onClick={onViewAllOptions}
+          >
+            <Icon type="arrow-left" />{' '}
```
Comment (binh-dam-ibigroup) [inline lib/components/narrative/narrative-itineraries-header.tsx:59]: Oops this one fell off the cracks, replace `{' '}` by writing `<Icon type=... withSpace />`.

File: lib/components/narrative/narrative-itineraries-header.tsx [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+      ) : (
+        <>
+          <div
+            style={{ flexGrow: 1 }}
+            title={intl.formatMessage({
+              id: 'components.NarrativeItinerariesHeader.titleText',
+              // FIXME: strange react-intl type errors
+              // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+              // @ts-ignore
+              values: {
```
Comment (binh-dam-ibigroup) [inline lib/components/narrative/narrative-itineraries-header.tsx:80]: The syntax for the imperative API is of the form
```
intl.formatMessage({ id: ... } { issueNum: ..., ... })
```

File: lib/components/narrative/narrative-itineraries-header.tsx [hunk 12]
```diff
@@ -0,0 +1,158 @@
...
+                issueNum: errors.length,
+                itineraryNum: itineraries.length,
+                pending: pending ? 'true' : 'false'
+              }
+            })}
+          >
+            <span style={{ marginRight: '10px' }}>
+              <FormattedMessage
+                id="components.NarrativeItinerariesHeader.resultText"
```
Comment (binh-dam-ibigroup) [inline lib/components/narrative/narrative-itineraries-header.tsx:89]: Revert conversion to double quotes in this file.

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
I have partially reviewed and need to look at more files changes, here are my comments so far.

File: lib/main.js [hunk 13]
```diff
@@ -0,0 +1,76 @@
+// import this polyfill in order to make webapp compatible with IE 11
```
Comment (binh-dam-ibigroup) [inline lib/main.js:1]: I'm curious whether this and `app.js` could be merged into a single file?

File: package.json [hunk 14]
```diff
@@ -85,11 +84,12 @@
     "pdfkit": "^0.12.1",
     "prop-types": "^15.6.0",
     "qs": "^6.3.0",
-    "react": "^16.9.0",
```
Comment (binh-dam-ibigroup) [inline package.json:88]: What is the reason for saying any React 16? (here are some features, such as hooks and <> that are not available at 16.0)

**Review by binh-dam-ibigroup — APPROVED** [review]
This looks really good! I have two minor questions, but I'm going to approve so someone else can review.

File: lib/main.js (hunk 13 above)
Comment (miles-grant-ibigroup) [reply lib/main.js:1]: I think it could, but didn't want to create any more vectors for possible regressions!

File: package.json (hunk 14 above)
Comment (miles-grant-ibigroup) [reply package.json:88]: I tried upgrading to version 17, but enzyme does not yet officially support React 17 (https://github.com/enzymejs/enzyme/issues/2429). However, we should upgrade to latest 16! Will do that now.

Comment (miles-grant-ibigroup) [reply package.json:88]: Done in 93d62e8cd2b63f2cfed5f4501b3ecf9a2f387029

**Comment by miles-grant-ibigroup** [comment]
With the flex PR merged, this PR is now ready to go! @philip-cline

**Comment by miles-grant-ibigroup** [comment]
I've disabled eslint on some difficult-to-type files. These files should definitely be typed as soon as possible, but I don't think it's a good idea to do so in this PR as it would result in expanding an already very large PR.

**Review by philip-cline — APPROVED** [review]
